### DISCUSSION TITLE
Idea for an option plugin that provides R2DBC Driver metrics collection

### DISCUSSION THREAD

**Discussion by linghengqian** [apache/skywalking#9229] [Ideas]
- R2DBC SPI is a common set of definitions that build R2DBC Drivers on asynchronous tcp, http, and other clients, distinguishing it from the JDBC specification as a low-level network transport protocol.

- For the R2DBC community, it is relatively reasonable to use R2DBC Proxy to intercept and collect metrics, because if R2DBC Proxy is left aside, it is a bit of a hassle to directly use R2DBC SPI to support specific R2DBC-driven metric collection. 

- I think https://github.com/apache/skywalking-java would be more user-friendly to R2DBC if an optional plugin could be provided? 

- R2DBC Proxy is located under https://github.com/r2dbc/r2dbc-proxy , and it has some use cases ( https://github.com/ttddyy/r2dbc-proxy-examples/tree/master/java-agent-example ) for integrating Java Agents, along with Spring Cloud Sleuth also exposes indicators through such tools, cf. https://github.com/spring-cloud/spring-cloud-sleuth/pull/1946 . 

- I'm not sure if anyone is working on a similar job.

**Comment by wu-sheng** [comment]
Hi, SkyWalking community is open to plugin contributions once the target framework is open source, and this plugin doesn't instrument classes out of your own source codebase.

According to this principle, r2dbc plugin is acceptable. You need to follow plugin dev doc and plugin test doc to prepare the pull request.

**Comment by linghengqian** [comment]
- I will close the current discussion because the design of https://github.com/r2dbc/r2dbc-spi as a specification has been stagnant for more than 1 year and may not represent the real future of the JDK.
